### PR TITLE
Make cmake modules consistent names

### PR DESCRIPTION
**Pull request by vhvb1989**
fixes: #1355 

I don't think this is required to be mentioned in the change log.

### CODE REVIEW

**Review by RickWinter — APPROVED** [review]

File: CMakeLists.txt [hunk 1]
```diff
@@ -55,13 +55,13 @@ if(BUILD_TESTING)
 endif()
 
 # compiler warning flags globally
-include(global_compile_options)
+include(AzureGlobalCompileOptions)
 
 # Documentation automation function
-include(${CMAKE_CURRENT_SOURCE_DIR}/cmake-modules/doxygen_common.cmake)
+include(AzureGlobalCompileOptions)
```
Comment (ahsonkhan) [inline CMakeLists.txt:61]: This doesn't seem like the right include. We are including AGCO twice.

Comment (ahsonkhan) [reply CMakeLists.txt:61]:
```suggestion
include(AzureDoxygen)
```

File: sdk/core/azure-core/CMakeLists.txt [hunk 2]
```diff
@@ -10,12 +10,12 @@ set(CMAKE_WINDOWS_EXPORT_ALL_SYMBOLS ON)
 
 list(APPEND CMAKE_MODULE_PATH "${CMAKE_CURRENT_SOURCE_DIR}/../../../cmake-modules")
 
-include(az_vcpkg)
-include(az_version)
-include(CodeCoverage)
-include(DefineTransportAdapter)
-include(doxygen_common)
-include(global_compile_options)
+include(AzureVcpkg)
```
Comment (ahsonkhan) [inline sdk/core/azure-core/CMakeLists.txt:13]: Missed the newly added `download_test_data.cmake` - PascalCase it. Also, let's add `Json` in the name.

@RickWinter do we want to prefix cmake modules that came from external sources with `Azure` as well (so `AzureDownloadJsonTestData`?

**Review by ahsonkhan — CHANGES_REQUESTED** [review]
Fixup cmakelist and update the name for the new module.

**Comment by ahsonkhan** [comment]
> I don't think this is required to be mentioned in the change log.

I agree. But to be certain, can you verify that we don't mention these in any markdown docs and that customers are not meant to include or call these directly in their projects or builds. If both those are true, no need for a CL entry since it isn't useful for the end-user.

**Review by ahsonkhan — APPROVED** [review]

**Pull request merged**